code-challenges.test.js

```diff
@@ -28,30 +37,119 @@ const colors2 = [
+  arr.sort(() => Math.random() - 0.5) //.sort() sorts the values in alphabetical order, but Math.random() - 0.5 allows for the shuffle of the array randomly. 
+  return arr
+}
+console.log(shuffleArray(colors1))
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
+describe("totalVotes", () => {
+  it("Returns a number that subtracts the upVotes by the downVotes.", () => {
+    expect(totalVotes(votes1)).toEqual(11)
+    expect(totalVotes(votes2)).toEqual(-31)
+  })
+})
+// Good Failure! - Complete
+
+const votes1 = { 
+  upVotes: 13, 
+  downVotes: 2
+}
 // Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
+const votes2 = { 
+  upVotes: 2, 
+  downVotes: 33 
+}
 // Expected output: -31
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//Function Name: totalVotes
+//Input: Objects
+//Output: Number
+//Process: Create a function called totalVotes. Use the argument object.valueName to get just the number from the object. Then subtract it to get the final answer. 
+
+const totalVotes = (object) => {
+  return object.upVotes - object.downVotes
+}
+
+// console.log(totalVotes(votes1))
+// console.log(totalVotes(votes2))
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("fixArray", () => {
+  it("Returns an array that should repeat words from previous two arrays, but only have the same words once on the new array.", () => {
+    expect(fixArray(...dataTypesArray1,...dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
 
 const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//Function Name: fixArray
+//Input: Two arrays
+//Output: One array
+//Process: Create a function called fixArray. Take in two arrays in the argument. Create a variable and within it use filter to filter through an a array and have it check what values we don't want duplicated. Return the value of everything that does not include "string" and "Boolean". Outside of the .filter return both arrays combined. 
+
+//First Try
+// Could not get it to run with JEST! Could you explain, please. 
+// const fixArray = (arr1, arr2) => {
+//    const filterArr = arr2.filter((value) => {
+//     return ! value.includes("string") && ! value.includes("Boolean")
```

Comment:
Here you are trying to hard code duplicates that you don't want in your array, but we always want our code to by dynamic! The main issue is that your test is performing a function call with the spread operator. You don't want the spread operator in the function call itself, only in the declaration of the function (when you are writing the actual function out). So the expect statement should look like this:
`expect(noDupes(dataTypesArray1, dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])`